From a4743b9b2b355f84da3904e7da93ec3c7a521895 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Fri, 26 Jul 2019 14:06:16 -0400
Subject: [PATCH] data: reap gdm sessions on shutdown

If GDM gets shutdown we should make sure all sessions get shutdown too.

This is a bit of a safety net in case any processes in the session are
lingering after the orderly shutdown.
---
 data/gdm.service.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/data/gdm.service.in b/data/gdm.service.in
index 17e8a8de..195bd0fd 100644
--- a/data/gdm.service.in
+++ b/data/gdm.service.in
@@ -1,33 +1,34 @@
 [Unit]
 Description=GNOME Display Manager
 
 # replaces the getty
 Conflicts=getty@tty${GDM_INITIAL_VT}.service
 After=getty@tty${GDM_INITIAL_VT}.service
 
 # replaces plymouth-quit since it quits plymouth on its own
 Conflicts=${PLYMOUTH_QUIT_SERVICE}
 After=${PLYMOUTH_QUIT_SERVICE}
 
 # Needs all the dependencies of the services it's replacing
 # pulled from getty@.service and ${PLYMOUTH_QUIT_SERVICE}
 # (except for plymouth-quit-wait.service since it waits until
 # plymouth is quit, which we do)
 After=rc-local.service plymouth-start.service systemd-user-sessions.service
 
 # GDM takes responsibility for stopping plymouth, so if it fails
 # for any reason, make sure plymouth still stops
 OnFailure=plymouth-quit.service
 
 [Service]
 ExecStart=${sbindir}/gdm
+ExecStopPost=-/usr/bin/bash -c 'for f in /run/systemd/sessions/*; do [ -f $f ] && /usr/bin/fgrep -q SERVICE=gdm $f && loginctl terminate-session $(basename $f); done'
 KillMode=mixed
 Restart=always
 IgnoreSIGPIPE=no
 BusName=org.gnome.DisplayManager
 EnvironmentFile=-${LANG_CONFIG_FILE}
 ExecReload=/bin/kill -SIGHUP $MAINPID
 KeyringMode=shared
 
 [Install]
 Alias=display-manager.service
-- 
2.32.0

